control-plane/connect-inject/endpoints_controller.go

```diff
@@ -818,7 +820,7 @@ func (r *EndpointsController) deleteACLTokensForServiceInstance(client *api.Clie
 			tokenPodName := strings.TrimPrefix(tokenMeta[TokenMetaPodNameKey], k8sNS+"/")
 
 			// If we can't find token's pod, delete it.
-			if tokenPodName == podName {
+			if tokenPodName == podName && tokenMeta[MetaKeyPodUID] == podUID {
```

Comment:
How is backward compatibility maintained with this logic? Assuming that customers upgrade their existing installation to the latest release, `tokenMeta[MetaKeyPodUID]` could be an empty string for pods that got created before pushing this change. Won't we miss deleting tokens of such pods/service instances in such cases?

Reply:
Many thanks for reviewing @Ganeshrockz and that's a great catch!

I've updated the logic to cater for backward compatibility.

https://github.com/hashicorp/consul-k8s/blob/2942824bd1154093afd059f213a001230812862b/control-plane/connect-inject/endpoints_controller.go#L822-L826

Please let me know what you think.

Reply:
Looks good to me